src/component/mxgraph/shape/gateway-shapes.ts

```diff
@@ -85,24 +83,14 @@ export class ExclusiveGatewayShape extends GatewayShape {
   }
 
   private addExclusiveGatewaySymbol(c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number): void {
-    const { xS, yS, wS, hS } = this.getScaledGeometry(x, y, w, h);
-
-    c.begin();
-    c.moveTo(xS + wS * 0.105, yS);
-    c.lineTo(xS + wS * 0.5, yS + hS * 0.38);
-    c.lineTo(xS + wS * 0.895, yS);
-    c.lineTo(xS + wS, yS + hS * 0.11);
-    c.lineTo(xS + wS * 0.6172, yS + hS * 0.5);
-    c.lineTo(xS + wS, yS + hS * 0.89);
-    c.lineTo(xS + wS * 0.895, yS + hS);
-    c.lineTo(xS + wS * 0.5, yS + hS * 0.62);
-    c.lineTo(xS + wS * 0.105, yS + hS);
-    c.lineTo(xS, yS + hS * 0.89);
-    c.lineTo(xS + wS * 0.3808, yS + hS * 0.5);
-    c.lineTo(xS, yS + hS * 0.11);
-    c.close();
-
-    c.fillAndStroke();
+    const canvas = this.configureCanvasForIcon(c, w, h, 0);
+    this.translateToStartingIconPosition(c, x, y, w, h);
```

Comment:
:warning: this should be done in the wrapper canvas, let's manage this in another refactoring PR